core/src/main/java/org/apache/accumulo/core/client/security/tokens/KerberosToken.java

```diff
@@ -51,11 +52,11 @@
    *          The user that is logged in
    */
   public KerberosToken(String principal) throws IOException {
-    requireNonNull(principal);
+    this.principal = requireNonNull(principal);
     final UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
-    checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
-    checkArgument(principal.equals(ugi.getUserName()), "Provided principal does not match currently logged-in user");
-    this.principal = ugi.getUserName();
+    if (AuthenticationMethod.KERBEROS == ugi.getAuthenticationMethod()) {
+      checkArgument(ugi.hasKerberosCredentials(), "Subject is not logged in via Kerberos");
+    }
```

Comment:
Thinking about this again, I need to update the javadoc on KerberosToken.

We're enforcing an assumption (silently) that when the token is instantiated, the Kerberos user is already logged in. Whereas, something like the following, could be something a user comes up with and would be surprised when it doesn't work.

```java
UserGroupInformation myUser = getUserBasedOnCustomLogic();
final Instance inst = new ZooKeeperInstance(clientConf);
final AuthenticationToken token = new KerberosToken();
myUser.doAs(new PrivilegedExceptionAction<Void>() {
  @Override public Void run() throws Exception {
    Connector conn = inst.getConnector(principal, token);
    // ...
  }
});
```
Let me also include javadoc update to try to make clear that the above circumstance would not work.